### ISSUE TITLE
Can we roll-back the Stata version requirement?

### ISSUE THREAD

**Issue by bquistorff** [gvegayon/parallel#92]
The Stata version required was recently changed from v11 -> v16 (9b02bd1, baf24ed). Was this a necessity of new changes in the source code or just the lack of availability of older Stata versions for building the *.mlib? I regularly work on Stata 15 and would like to roll it back to at least there (and I have access to older versions if we want move it back even farther). Is there any noticeable benefit of compiling the mlib in the newer formats? If we roll it back, I can be in charge of building the *.mlibs.

**Comment by gvegayon** [comment]
No, there are no noticeable benefits; you can roll it back. There is an issue, though, regarding providing support for earlier versions of Stata. I posted a question a couple of months ago regarding this, and [a few ideas came](https://www.statalist.org/forums/forum/general-stata-discussion/general/1613670-updating-stata-journal-packages-and-ssc-the-parallel-module?p=1614361#post1614361). What do you think about it? I can think of two options:

1. Keep tagged versions like [parallel version-[stata version], e.g., `v1.20.1-stata15`
2. Keep everything under the same tag, e.g., v`1.20.1`, but with different compiled versions of it in subfolders.

**Comment by bquistorff** [comment]
I think managing releases for multiple versions is a lot given this package doesn't experience much change. I'd be happy if we just release a version that we (maintainers) can easily build and which serves most users (I'd vote for v15 or v14). There's probably a perfect solution for the problem, but it's probably not worth our time. I've run into several several pain-points with packages that automatically recompile. Then I think we make it possible to compile on older Stata's and just give the instructions (which I think we basically already do as one would just "download repo", run `do compile_and_install.do`, though we'd want to check). How does that seem?

**Comment by gvegayon** [comment]
That sounds good. I see how automatic compilation can be troublesome. We still haven't released a "new version" in Boston's SSC. If you are OK with releasing for state 15, let's do that, and I can start preparing the package for SSC.

**Comment by bquistorff** [comment]
I have a new feature that I've been prototyping. Could I get your feedback on it generally and whether we should try to include it before releasing in Boston's SSC? The idea was to make it easier to get the same results across `simulate`, `bootstrap`, and `permute` regardless of whether it was executed sequential or on an arbitrary number of clusters with `parallel`. The code is in the [reproducible_cmds branch](https://github.com/gvegayon/parallel/tree/reproducible_cmds). It builds off of the approach outlined in the wiki [Reproducible results with rngs](https://github.com/gvegayon/parallel/wiki/Gallery#reproducible-results-with-rngs) and sets up `simulate` to run things. For usage you can see the [test file](https://github.com/gvegayon/parallel/blob/reproducible_cmds/test%20files/test_seeding.do). Basically you can do
```
seeding simulate/bootstrap/permute, reps(#) parallel: cmd
```
which runs it parallel or you can drop `,parallel` and it will run sequentially.

There's still a bit more to do: more work on `seeding bootstrap` and `seeding permute` since they do some post-processing of the Monte Carlos results, check that options that are different between the 3 main commands mostly work, and it still needs a help file written.

**Comment by bquistorff** [comment]
Just added building instructions to the README.

**Comment by gvegayon** [comment]
Thanks, Brian! That new feature sounds like an excellent idea. Sure, let's add that before submitting the new version to SSC. I can't check it out right away, but I will next week.

**Comment by gvegayon** [comment]
Thanks for the patience! Everything looks good. The only thing I would add is a reference in the main manual of the package to the new function. Once we have that, I believe we are good to go! Thanks!
